mobile-widgets/qmlmanager.cpp

```diff
@@ -869,13 +869,11 @@ void QMLManager::commitChanges(QString diveId, QString date, QString location, Q
 		d->suit = strdup(qPrintable(suit));
 	}
 	if (myDive->buddy() != buddy) {
-		if (myDive->buddy().contains(",")) {
-			if (!buddy.contains(tr("Multiple Buddies"))) {
-				diveChanged = true;
-				free(d->buddy);
-				d->buddy = strdup(qPrintable(buddy));
-			}
-		} else {
+		if (buddy.contains(",") && !buddy.contains(QRegExp("(,\\s)"))){
```

Comment:
In order for the outcome to be correct the string that is actually saved to the logbook needs to have a space after the comma or they will be saved as one. In your example above you would get two buddies, "Linus,Joe" and "John".
So we might end up having to split on all commas then add a check to see if the first character is a space or not, if it is not a space add one when reassembling the string.